src/views/WorkspaceDetail.vue

```diff
@@ -143,107 +163,109 @@ const workspaceNameRules: Array<(x: string) => string|boolean> = [
       ];
 
       return errors.filter((res): res is string => typeof res === 'string');
-    },
-  },
-
-  watch: {
-    workspace() {
-      this.update();
-    },
+    });
 
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    localWorkspace(this: any) {
-      // Once the user types, clears the error returned on sending the rename API call.
-      this.requestError = null;
-    },
-  },
-  created() {
-    this.update();
-  },
-  methods: {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    cancelRename(this: any) {
-      this.requestError = null;
-      this.localWorkspace = this.workspace;
-      this.editing = false;
-    },
+    function cancelRename() {
+      requestError.value = null;
+      localWorkspace.value = props.workspace;
+      editing.value = false;
+    }
 
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    async renameWorkspace(this: any) {
-      if (this.nameErrorMessages.length) {
+    async function renameWorkspace() {
+      if (nameErrorMessages.value.length) {
         return;
       }
 
-      if (this.localWorkspace === this.workspace) {
-        this.editing = false;
+      if (localWorkspace.value === props.workspace) {
+        editing.value = false;
         return;
       }
 
-      if (this.localWorkspace !== null) {
+      if (localWorkspace.value !== null) {
         try {
-          const { name } = await api.renameWorkspace(this.workspace, this.localWorkspace);
-          this.$router.push(`/workspaces/${name}`);
-          this.editing = false;
-          this.requestError = null;
+          const { name } = await api.renameWorkspace(props.workspace, localWorkspace.value);
+          router.push(`/workspaces/${name}`);
+          editing.value = false;
+          requestError.value = null;
 
           store.dispatch.fetchWorkspaces();
         } catch (err) {
           if (err.response.status === 409) {
-            this.requestError = 'A workspace by that name already exists';
+            requestError.value = 'A workspace by that name already exists';
           } else {
-            this.requestError = `${Object.values(err.response.data).flat()[0]}`;
+            requestError.value = `${Object.values(err.response.data).flat()[0]}`;
           }
         }
       }
-    },
+    }
 
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    async update(this: any) {
-      this.loading = true;
+    async function update(this: any) {
+      loading.value = true;
 
-      this.localWorkspace = this.workspace;
-      await store.dispatch.fetchWorkspace(this.workspace);
-      this.loading = false;
-    },
-  },
+      localWorkspace.value = props.workspace;
+      await store.dispatch.fetchWorkspace(props.workspace);
+      loading.value = false;
+    }
+
+    watch(() => props.workspace, () => update());
+    // eslint-disable-next-line no-return-assign
+    watch(localWorkspace, () => requestError.value = null);
```

Comment:
Rather than ignore the rule, I'd just follow it here

```suggestion
    watch(localWorkspace, () => { requestError.value = null; });
```